src/trie/NibbleVec.sol

```diff
@@ -0,0 +1,80 @@
+pragma solidity ^0.8.17;
+
+// SPDX-License-Identifier: Apache2
+
+struct NibbleVec {
+    bytes data;
+}
+
+library NibbleVecOps {
+    function fromHex(
+        bytes memory hexBytes
+    ) internal pure returns (NibbleVec memory) {
+        return NibbleVec(hexBytes);
+    }
+
+    function isLeaf(NibbleVec memory self) internal pure returns (bool) {
+        return self.data[self.data.length - 1] == bytes1(uint8(16));
+    }
+
+    function fromRaw(
+        bytes memory rawBytes,
+        uint8 is_leaf
+    ) internal pure returns (NibbleVec memory) {
+        require(is_leaf == 0 || is_leaf == 1, "is_leaf value not accepted");
+        uint size = rawBytes.length * 2 + is_leaf;
+        bytes memory hexBytes = new bytes(size);
+        uint j = 0;
+        for (uint256 i = 0; i < rawBytes.length; i++) {
+            hexBytes[j] = bytes1(uint8(rawBytes[i]) / 16);
+            hexBytes[j + 1] = bytes1(uint8(rawBytes[i]) % 16);
+            j = j + 2;
+        }
+        if (is_leaf == 1) {
+            hexBytes[j] = bytes1(uint8(16));
+        }
+        return NibbleVec(hexBytes);
+    }
+
+    function fromCompact(
+        bytes memory compactBytes
+    ) internal pure returns (NibbleVec memory) {
+        uint8 flag = uint8(compactBytes[0]);
+        require(flag & 192 == 0, "reserved flag bits used");
+        uint8 prefix = flag >> 4;
+        //uint8 is_odd = prefix & 1 == 1 ? 1 : 0;
+        uint8 is_leaf = prefix & 2 == 2 ? 1 : 0;
+        uint size = (compactBytes.length - 1) * 2 + is_leaf /*+ is_odd*/;
+        bytes memory hexBytes = new bytes(size);
+        uint j = 0;
+        // if (is_odd == 1) {
+        //     hexBytes[j] = bytes1(flag % 16);
+        //     j = j + 1;
+        // }
+        for (uint256 i = 1; i < compactBytes.length; i++) {
+            hexBytes[j] = bytes1(uint8(compactBytes[i]) / 16);
+            hexBytes[j + 1] = bytes1(uint8(compactBytes[i]) % 16);
+            j = j + 2;
+        }
```

Comment:
is this hex decoding?

Reply:
It should be hex encoding (or better, representation), because here we are representing each byte with two bytes, while in the reverse function (e.g., `encodeCompact`, which is not included because it was not useful for the overall implementation, but the same thing applies to `fromRaw` and `encodeRaw`) we rebuild a single byte from the hexadecimal representation.

Reply:
I'm a bit confused as to why this is needed. This makes the whole ethereum proof verification super expensive

Reply:
I followed the implementation of this [cita-trie](https://github.com/citahub/cita-trie) library. Let me do some further testing without these transformations, maybe it is not necessary.

Reply:
I confirm that this is actually not needed, pushing the fixed version in a while. Thanks for noticing!